chat/views.py

```diff
@@ -1,3 +1,64 @@
-# from django.shortcuts import render
+from django.shortcuts import render
+from django.core.exceptions import ValidationError
+from chat.models import Message
+from chat.serializer import MessageSerializer
+from course.models import StudentCourse
+from study_group.models import StudyGroup
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.decorators import api_view
+
 
 # Create your views here.
+def chat_view(request):
+    if request.user.profile.account_type == 'S':
+        student_courses = StudentCourse.objects.get_student_courses(request.user)
+    else:
+        student_courses = StudentCourse.get_student_courses_by_teacher(request.user)
+    study_groups = StudyGroup.objects.get_groups_by_user(request.user)
+    context = {
+        'student_courses': student_courses,
+        'study_groups': study_groups,
+    }
+    return render(request, 'chat/chat.html', context)
+
+
+@api_view(['GET'])
+def getCourseChat(request, course_id):
+    course = StudentCourse.objects.get_course(course_id)[0]
+    if not request.user.is_authenticated or (
+            request.user != course.student_id and request.user != course.teacher_course_id.teacher_id):
+        return Response(status=status.HTTP_401_UNAUTHORIZED)
+    messages = Message.objects.get_student_course_chat(student_course_id=course_id)
+    serializer = MessageSerializer(messages, many=True)
+    return Response(serializer.data)
+
+
+@api_view(['GET'])
+def getGroupChat(request, group_id):
+    group = StudyGroup.objects.get_group_by_id(group_id)[0]
+    if not request.user.is_authenticated or not group.is_user_in_group(request.user):
+        return Response(status=status.HTTP_401_UNAUTHORIZED)
+    messages = Message.objects.get_group_chat(group_id=group_id)
+    serializer = MessageSerializer(messages, many=True)
+    return Response(serializer.data)
+
+
+@api_view(['POST'])
+def addMessage(request):
+    if not request.user.is_authenticated:
+        return Response(status=status.HTTP_401_UNAUTHORIZED)
+    if 'courseId' in request.data:
+        course = StudentCourse.objects.get_course(int(request.data['courseId']))[0]
+    else:
+        course = None
```

Comment:
You can also declare the course variable before the conditional statement, removing the need for the `else`
Also, is trying to get `StudentCourse.objects.get_course(int(request.data['courseId']))` fails when there is no `courseId` in request.data? or does it just return None? if so, you can remove the whole if (or add a `or none` at the end of the assignment).

Reply:
Changed (removed the else clause). if 'courseId' is not in request.data, the above call will raise an exception (when calling request.data['courseId']), so the if is still needed. Thanks to this comment I noticed that get_course can return an empty list (if it contains bad courseId), so I modified it to check for this case.